### PR TITLE
Streamline the Repository Interface and add docu

### PR DESCRIPTION
**Pull request by white-gecko**
Improve Repository interface and documentation.

(IMO: Ready to merge)

### CODE REVIEW

File: quit/git.py [hunk 1]
```diff
@@ -22,15 +22,30 @@
...
+    """
+
+    def __init__(self, path, origin=None, create=False, garbageCollection=False):
+        """Initialize a quit repo at a given location of the filesystem.
+
+        Keyword arguments:
+        path -- the location of an existing git repository or where the repository should be created
+        origin -- if the repository does not exist and origin is given the origin repository is
+                  cloned to the given path (default: None)
+        create -- boolean whether to create a new repository if the cpecified path is empty
```
Comment (splattater) [inline quit/git.py]: typo  cpecified -> specified

File: quit/git.py [hunk 2]
```diff
@@ -22,15 +22,30 @@
...
+                  (default: False)
+        garbageCollection -- boolean whether to activate the garbage collection on the git
+                  repository (default: False)
+        """
         self.callback = QuitRemoteCallbacks()
 
         try:
             self._repository = pygit2.Repository(path)
         except KeyError:
```
Comment (splattater) [inline quit/git.py]: Do you know if KeyError is raised in your case?

Comment (white-gecko) [reply quit/git.py]: This is not subject of this PR

File: quit/git.py (hunk 1 above)
Comment (white-gecko) [reply quit/git.py]: DONE

**Pull request merged**